### PR TITLE
SRCH-2472 re-record cassettes for specs

### PR DESCRIPTION
**Pull request by MothOnMars**
## Summary
This PR:
- removes unused cassettes
- updates the VCR & webmock gems

I've also updated a couple of specs to use Webmock instead of VCR, and to ensure that the spec passes with the updated recording. As always, all the files in `spec/vcr_cassettes` are auto-generated and do not require individual review.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional.

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock. - N/A
 
- [x] You have merged the latest changes from the target branch (usually `master` or `main`) into your branch.
 
- [x] If your target branch is NOT `master` or `main`, specify the reason:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- [x] You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop)

- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -40,4 +40,5 @@ exclude_patterns:
 - vendor/
 - app/api/search_consumer/
 -  "**/vendor/**/*"
+-  "**/vcr_cassettes/**/*"
```
Comment (jmax-fearless) [inline .codeclimate.yml:43]: Question only: Why is this necessary? The vcr_cassettes are YAML files, which CC shouldn't be checking anyway.

**Review by jmax-fearless — APPROVED** [review]
LGTM

(I'm presuming that the two actual spec files that you changed needed to be updated to match the newly recorded cassettes).

File: .codeclimate.yml (hunk 1 above)
Comment (MothOnMars) [reply .codeclimate.yml:43]: Code Climate runs 8 other engines in addition to Rubocop (see the list of `plugins`: https://github.com/GSA/search-gov/blob/master/.codeclimate.yml#L8). Some of those will check YAML files. For example, the `fixme` engine will flag stray `# TODO` or `# FIXME` comments in those files.

File: Gemfile [hunk 2]
```diff
@@ -168,8 +168,8 @@ group :test do
   gem 'poltergeist', '~> 1.18.1'
   gem 'shoulda-matchers', '~> 4.1.1'
   gem 'shoulda-kept-assign-to', '~> 1.1.0'
-  gem 'vcr', '~> 4.0'
-  gem 'webmock', '~> 3.8.3'
+  gem 'vcr', '~> 6.0'
```
[the remark this answers is not in the document]
Comment (MothOnMars) [reply Gemfile:171]: Not necessary - I just took the opportunity to update the gems that I was working with. I won't be running `bundle update` for this story, as these changes won't be manually tested. The only gem updates are for gems that are only used in a test environment.
